### PR TITLE
[System.Text.Json] Fix bug complex properties

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
fixes https://github.com/dotnet/corefx/issues/37536
fixes https://github.com/dotnet/corefx/issues/37567

Suggest me more test case if needed.

cc: @steveharter @ahsonkhan

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
unrelated ci fail https://mc.dot.net/#/user/dotnet-bot/pr~2Fdotnet~2Fcorefx~2Frefs~2Fpull~2F37586~2Fmerge/test~2Ffunctional~2Fcli~2F/20190510.15

File: src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleNull.cs [hunk 1]
```diff
@@ -25,6 +25,12 @@ private static bool HandleNull(ref Utf8JsonReader reader, ref ReadStack state, J
             }
 
             JsonPropertyInfo propertyInfo = state.Current.JsonPropertyInfo;
+
+            if (!propertyInfo.ShouldDeserialize)
+            {
+                return false;
```
Comment (MarcoRossignoli) [inline src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleNull.cs]: If I understood well we should return true only if we're sure to be at the end of json object root

**Comment by ahsonkhan** [comment]
> fixes #37567

> Suggest me more test case if needed.

I would add a test with properties that have private setters like the one that was stated in the original issue (both for top-level properties, and for complex properties/nested).

**Comment by MarcoRossignoli** [comment]
Close because I confirm that update https://github.com/dotnet/corefx/pull/37578 fix bug my tests passed. cc: @steveharter @ahsonkhan

**Comment by ahsonkhan** [comment]
> Close because I confirm that update https://github.com/dotnet/corefx/pull/37578 fix bug my tests passed. 

Both of them? #37567 as well?

@steveharter consider adding the test cases from @MarcoRossignoli from this PR into yours.

**Comment by MarcoRossignoli** [comment]
> Both of them? #37567 as well?

I think so I did some test with @steveharter clone and sample on #37567 works

**Pull request closed without merging**